core/src/test/scala/com/mesosphere/usi/core/scalaapi/SchedulerAdapterTest.scala

```diff
@@ -0,0 +1,315 @@
+package com.mesosphere.usi.core.scalaapi
+
+import akka.NotUsed
+import akka.stream.scaladsl.{SinkQueueWithCancel, SourceQueueWithComplete}
+import akka.stream.scaladsl.{Flow, Sink, Source}
+import akka.stream.OverflowStrategy
+import com.mesosphere.usi.core.Scheduler
+import com.mesosphere.usi.core.models.resources.ScalarRequirement
+import com.mesosphere.usi.core.models._
+import com.mesosphere.utils.AkkaUnitTest
+import org.scalatest._
+
+import scala.concurrent.Promise
+import scala.util.{Failure, Success}
+
+class SchedulerAdapterTest extends AkkaUnitTest with Inside {
+
+  def createMockedScheduler: (
+      Flow[Scheduler.SpecInput, Scheduler.StateOutput, NotUsed],
+      SinkQueueWithCancel[Scheduler.SpecInput],
+      SourceQueueWithComplete[Scheduler.StateOutput]) = {
+
+    val (outerSinkQueue, outerSink) = Sink.queue[Scheduler.SpecInput]().preMaterialize()
+    val (outerSourceQueue, outerSource) =
+      Source.queue[Scheduler.StateOutput](16, OverflowStrategy.fail).preMaterialize()
+
+    (Flow.fromSinkAndSource(outerSink, outerSource), outerSinkQueue, outerSourceQueue)
+  }
+
+  class SchedulerFixture {
+    val podSpec = PodSpec(
+      PodId("podId"),
+      Goal.Running,
+      RunSpec(
+        resourceRequirements = List(ScalarRequirement.cpus(1), ScalarRequirement.memory(256)),
+        shellCommand = "sleep 3600")
+    )
+    val specEvent = PodSpecUpdated(podSpec.id, None)
+    val stateEvent = PodStatusUpdated(podSpec.id, None)
+
+    val (scheduler, specInputQueue, stateOutputQueue) = createMockedScheduler
+    val adapter = new SchedulerAdapter(scheduler)
+  }
+
+  "SchedulerAdapter" when {
+    "materialized as Source and Sink" should {
+      "Propagate a snapshot to the underlying scheduler" in new SchedulerFixture {
+        adapter.asSourceAndSink(SpecsSnapshot(List(podSpec), Nil))
+        specInputQueue.pull().futureValue.get._1.podSpecs.head shouldEqual podSpec
+      }
+
+      "If snapshot is not provided push an empty snapshot" in new SchedulerFixture {
+        adapter.asSourceAndSink()
+        specInputQueue.pull().futureValue.get._1 shouldEqual SpecsSnapshot.empty
```

Comment:
```suggestion
        specInputQueue.pull().futureValue.get._1 should be('empty)
```

We may have to add `isEmpty` to `SpecsSnapshot`.

Also, shouldn't this be a state snapshot?

Reply:
It's actually

```
 specInputQueue.pull().futureValue.get._1 shouldBe empty
```

Reply:
http://www.scalatest.org/user_guide/using_matchers

Reply:
I feel like defining an implicit value that extend `trait Emptiness[-T]` is a bit overkill for this case.